fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBDatabaseFactory.java

```diff
@@ -121,6 +122,13 @@
      */
     @Nonnull
     private Supplier<Boolean> transactionIsTracedSupplier = LOGGER::isTraceEnabled;
+    /**
+     * A supplier that, when it returns true, enables assertions/sanity checks on certain operations. This supplier is
+     * polled upon the creation of a new {@code FDBRecordContext} and all subsequent operations on that context will
+     * (or will not) enforce assertions based upon the return value.
+     */
+    @Nonnull
+    private Supplier<Boolean> enableAssertionsSupplier = DEFAULT_ENABLE_ASSERTIONS_SUPPLIER;
```

Comment:
Hmm, I'm not sure this is necessarily the best way to determine whether or not to enable assertions or not. In particular, it wouldn't let you do something like set it for a single transaction (if, for example, if one wanted to turn on "debugging" for a single transaction that both, like, traced it and enabled assertions). Like, the other mechanism I can think of would be to add a new field to the `FDBRecordContextConfig` (and, in theory, one could want that for tracing). I suppose it's possible that we could go with this "for now", see how it goes, and add a new mechanism, if we don't like the way that that works in practice.

Reply:
Yeah, you are absolutely right. I had forgotten about that object. I have moved the configuration there.